src/core/annotation.js

```diff
@@ -146,6 +146,39 @@ class AnnotationFactory {
   }
 }
 
+function getQuadPoints(dict, rect) {
+  if (!dict.has('QuadPoints')) {
+    return null;
+  }
+
+  // The region is described as a number of quadrilaterals.
+  // Each quadrilateral must consist of eight coordinates.
+  const quadPoints = dict.getArray('QuadPoints');
+  if (!Array.isArray(quadPoints) || quadPoints.length % 8 > 0) {
+    return null;
+  }
+
+  const quadPointsLists = [];
```

Comment:
Nit: You could probably initialize this Array with the desired format right away, but I don't know how much it matters; e.g. something like this:
```javascript
const numQuadPoints = quadPoints.length / 8;
const quadPointsLists = new Array(numQuadPoints).fill([]);
for (let i = 0; i < numQuadPoints; i++) {
  ...
```

Reply:
I tried this, but it doesn't work because `fill` does not really work well with objects. From the MDN docs: `When fill gets passed an object, it will copy the reference and fill the array with references to that object.` This has the undesirable effect that pushing to one of the arrays pushes the value to all of them, which made the unit tests fail. According to https://medium.com/@wisecobbler/4-ways-to-populate-an-array-in-javascript-836952aea79f we could use the `fill` approach with `map`, but that looks a bit strange and also copies the whole array again, so I don't think there is an advantage in doing that.